src/device/cuda/cooperative_groups.jl

```diff
@@ -1,42 +1,71 @@
 # C. Cooperative Groups
 
-export this_grid, sync_grid
+export this_grid, sync_grid,
+       this_multi_grid, sync_multi_grid
 
-"""
-    this_grid()
+@enum cudaCGScope::Cuint begin
+      cudaCGScopeInvalid   = 0  # Invalid cooperative group scope
+      cudaCGScopeGrid           # Scope represented by a grid_group
+      cudaCGScopeMultiGrid      # Scope represented by a multi_grid_group
+end
 
-Returns a `grid_handle` of the grid group this thread belongs to. Only available if a cooperative
-kernel is launched.
-"""
-this_grid
 
+# Both grid groups and multi-grid groups use the same intrinsic to get handle:
 if VERSION >= v"1.2.0-DEV.512"
-    @inline this_grid() = ccall("extern cudaCGGetIntrinsicHandle", llvmcall, Culonglong, (Cuint,), UInt32(1))
+    @inline _get_handle(cooperative_scope::cudaCGScope) =
```

Comment:
maybe rename this to `cg_intrinsic_handle`; without namespacing this could clash or get confused with other handle functions